sycl-jit/jit-compiler/lib/KernelFusion.cpp

```diff
@@ -235,6 +236,31 @@ extern "C" JITResult fuseKernels(View<SYCLKernelInfo> KernelInformation,
   return JITResult{FusedKernelInfo};
 }
 
+extern "C" JITResult compileSYCL(const char *SYCLSource,
+                                 View<IncludePair> IncludePairs,
+                                 View<const char *> UserArgs,
+                                 const char *DPCPPRoot) {
+  std::unique_ptr<llvm::Module> Module =
+      compileDeviceCode(SYCLSource, IncludePairs, UserArgs, DPCPPRoot);
+  if (!Module) {
+    return JITResult{"Device code compilation failed"};
+  }
+
+  SYCLKernelInfo Kernel;
+  auto Error = translation::KernelTranslator::translateKernel(
+      Kernel, *Module, JITContext::getInstance(), BinaryFormat::SPIRV);
+
+  auto *LLVMCtx = &Module->getContext();
+  Module.reset();
+  delete LLVMCtx;
```

Comment:
How expensive is it to set up and destroy the `LLVMContext` on every call to RTC? Would it be an alternative to use the context from `JITContext`?

Reply:
If I'm not mistaken, certain things like constants and metadata are stored within `LLVMContext` and won't be deallocated even if a module which references them is deallocated. Therefore, keeping `LLVMContext` between RTC call invocations could result in some memory build-up.

At least that is the behavior we discovered a few years ago when we were debugging exceptionally huge memory footprint of `sycl-post-link` where we had a huge codebase compiled with per-kernel device code split. I don't know if anything has changed since that in the upstream LLVM, but we hadn't proposed any patches back then.